walker.go

```diff
@@ -192,9 +249,104 @@ func (s *StatInfo) Sys() interface{} {
 	return s.Stat
 }
 
-func matchPrefix(pattern, name string) (bool, bool) {
-	count := strings.Count(name, string(filepath.Separator))
+func matchPrefix(pattern, name string, isDir bool) (bool, bool) {
 	partial := false
+	if strings.Contains(pattern, "**") {
+		// short-circuit for single "**"
+		if pattern == "**" || pattern == "**/" {
+			return true, isDir
+		}
+		pattParts := strings.Split(pattern, string(filepath.Separator))
```

Comment:
This looks like a function better avoided in a function called for every scanned file(times number of patterns). Although I guess `filepath.Walk` isn't very memory efficient either.

Reply:
Yeah, it is probably not the most efficient, but at least the cost is only paid when `**` is used, otherwise it will be the same cost (plus a `strings.Contains`) as before.  

I can look into adding some benchmarking tests to try to see what the relative cost is, and perhaps set a goal to beat in the future.